### PR TITLE
Reduce the restrictiveness of the filename filter in base

### PR DESCRIPTION
**Pull request by chrisdonaldson**
Fixes issue #9 and bumps the version number to v2.2.0.

@jamesmacaulay @karlhungus @bouk

### CODE REVIEW

**Comment by karlhungus** [comment]
Related: https://github.com/Shopify/asset_cloud/pull/10

from the previous pr :fries:

**Comment by jamesmacaulay** [comment]
LGTM :+1:

**Comment by chrisdonaldson** [comment]
I'll wait for @bouk to comment before merging. I have run a couple thousand random real asset_cloud assets through the new regex without any failures.

File: lib/asset_cloud/base.rb [hunk 1]
```diff
@@ -9,7 +9,25 @@ class IllegalPath < StandardError
   class Base
     cattr_accessor :logger
 
-    VALID_PATHS = /\A[a-z0-9][a-z0-9_\-\/]+([a-z0-9][\w\-\ \.\@]*\.\w{2,6})?\z/i
+    VALID_PATHS = /\A
+      (
+        ([\w])                #Filename can be a single letter or underscore
```
Comment (bouk) [inline lib/asset_cloud/base.rb]: I don't think you need `[]` around `\w`

Comment (chrisdonaldson) [inline lib/asset_cloud/base.rb]: Nope. Relic of having restricted starting with an underscore. I'll remove them.

File: spec/base_spec.rb [hunk 2]
```diff
@@ -42,8 +42,39 @@ class BasicCloud < AssetCloud::Base
     lambda { @fs['./test']   }.should raise_error(AssetCloud::IllegalPath)
   end
 
+  it "should raise error when filename has trailing period" do
+    lambda { @fs['test.']           }.should raise_error(AssetCloud::IllegalPath)
+    lambda { @fs['/test/testfile.'] }.should raise_error(AssetCloud::IllegalPath)
+    lambda { @fs['test/directory/.']}.should raise_error(AssetCloud::IllegalPath)
```
Comment (bouk) [inline spec/base_spec.rb]: there's some tests missing here, ending with space, ending with slash, ending with slash dot

Comment (chrisdonaldson) [inline spec/base_spec.rb]: Agreed.

**Comment by bouk** [comment]
lgtm

**Comment by chrisdonaldson** [comment]
I had concerns last night that there might be files that were caught by the new regex that were not caught by the old one. I pulled a much larger sample size from a live asset_cloud (1,000,000 records) and found that multiple dots are frequently used as well as brackets. The latest commit makes both of those allowable without interfering with the previous reduction in restrictiveness.

**Pull request merged**